diff -uprN ../node-zwave-js-9.0.6-original/packages/config/config/devices/0x0109/zl7432.json ./packages/config/config/devices/0x0109/zl7432.json
--- ../node-zwave-js-9.0.6-original/packages/config/config/devices/0x0109/zl7432.json	2022-05-01 15:19:30.952876874 -0700
+++ ./packages/config/config/devices/0x0109/zl7432.json	2022-05-01 15:11:31.503113633 -0700
@@ -25,5 +25,8 @@
 			"maxNodes": 5,
 			"isLifeline": true
 		}
+	},
+	"compat": {
+		"forceRefreshValuesCCReport": true
 	}
 }
diff -uprN ../node-zwave-js-9.0.6-original/packages/config/src/devices/CompatConfig.ts ./packages/config/src/devices/CompatConfig.ts
--- ../node-zwave-js-9.0.6-original/packages/config/src/devices/CompatConfig.ts	2022-05-01 15:19:30.972877071 -0700
+++ ./packages/config/src/devices/CompatConfig.ts	2022-05-01 15:13:30.711311028 -0700
@@ -99,6 +99,19 @@ error in compat option enableBasicSetMap
 			this.enableBasicSetMapping = definition.enableBasicSetMapping;
 		}
 
+		if (definition.forceRefreshValuesCCReport != undefined) {
+			if (definition.forceRefreshValuesCCReport !== true) {
+				throwInvalidConfig(
+					"devices",
+					`config/devices/${filename}:
+error in compat option forceRefreshValuesCCReport`,
+				);
+			}
+
+			this.forceRefreshValuesCCReport =
+				definition.forceRefreshValuesCCReport;
+		}
+
 		if (definition.forceNotificationIdleReset != undefined) {
 			if (definition.forceNotificationIdleReset !== true) {
 				throwInvalidConfig(
@@ -468,6 +481,7 @@ compat option alarmMapping must be an ar
 	public readonly disableBasicMapping?: boolean;
 	public readonly disableStrictEntryControlDataValidation?: boolean;
 	public readonly enableBasicSetMapping?: boolean;
+	public readonly forceRefreshValuesCCReport?: boolean;
 	public readonly forceNotificationIdleReset?: boolean;
 	public readonly forceSceneControllerGroupCount?: number;
 	public readonly manualValueRefreshDelayMs?: number;
diff -uprN ../node-zwave-js-9.0.6-original/packages/zwave-js/src/lib/node/Node.ts ./packages/zwave-js/src/lib/node/Node.ts
--- ../node-zwave-js-9.0.6-original/packages/zwave-js/src/lib/node/Node.ts	2022-05-01 15:19:30.980877150 -0700
+++ ./packages/zwave-js/src/lib/node/Node.ts	2022-05-01 15:17:01.795563195 -0700
@@ -51,6 +51,7 @@ import {
 	stringify,
 	TypedEventEmitter,
 } from "@zwave-js/shared";
+import { BinarySwitchCCReport } from "../commandclass/BinarySwitchCC";
 import { roundTo } from "alcalzone-shared/math";
 import { padStart } from "alcalzone-shared/strings";
 import { isArray, isObject } from "alcalzone-shared/typeguards";
@@ -2306,6 +2307,8 @@ protocol version:      ${this.protocolVe
 			return this.handleFirmwareUpdateStatusReport(command);
 		} else if (command instanceof EntryControlCCNotification) {
 			return this.handleEntryControlNotification(command);
+		} else if (command instanceof BinarySwitchCCReport) {
+			return this.handleBinarySwitchReport(command);
 		} else if (command instanceof PowerlevelCCTestNodeReport) {
 			return this.handlePowerlevelTestNodeReport(command);
 		}
@@ -2331,6 +2334,16 @@ protocol version:      ${this.protocolVe
 
 	private hasLoggedNoNetworkKey = false;
 
+	private handleBinarySwitchReport(_command: BinarySwitchCCReport): void {
+		if (this.deviceConfig?.compat?.forceRefreshValuesCCReport) {
+			this.driver.controllerLog.logNode(
+				this.id,
+				`Refreshing values for all endpoints...`,
+			);
+			void this.refreshValues();
+		}
+	}
+
 	/**
 	 * @internal
 	 * Handles a nonce request
